04-injection-part-1/README.md

```diff
@@ -434,22 +375,59 @@ Most instances of SQL injection can be prevented by using parameterized queries
 
-`PreparedStatement statement = connection.prepareStatement("SELECT * FROM products WHERE category = ?");`
+# Tools
 
-`statement.setString(1, input);`
+## Burp Suite
+[Burp Suite][1] is one of the most popular penetration testing and vulnerability finder tools, and is often used for checking web application security. With this tool you can see all the requests that are executed when interacting with a web application. You can change the requests to see different behaviors and you can set automatic requests with different parameters. [[1]]
 
-`ResultSet resultSet = statement.executeQuery();`
+## SQLmap
+[SQLmap][7] is an open source penetration testing tool that automates the process of detecting and exploiting SQL injection flaws and taking over of database servers. It comes with a powerful detection engine, many niche features for the ultimate penetration tester and a broad range of switches lasting from database fingerprinting, over data fetching from the database, to accessing the underlying file system and executing commands on the operating system via out-of-band connections. You can see some of the features in this recording. [[7]]
+[![asciicast](https://asciinema.cs.pub.ro/a/N1r0DeFboLW4GXj0rZb4jsYeU.png)](https://asciinema.cs.pub.ro/a/N1r0DeFboLW4GXj0rZb4jsYeU)
 
-Parameterized queries can be used for any situation where untrusted input appears as data within the query, including the WHERE clause and values in an INSERT or UPDATE statement. They can't be used to handle untrusted input in other parts of the query, such as table or column names, or the ORDER BY clause. Application functionality that places untrusted data into those parts of the query will need to take a different approach, such as white-listing permitted input values, or using different logic to deliver the required behavior.
+# Further Reading
 
-For a parameterized query to be effective in preventing SQL injection, the string that is used in the query must always be a hard-coded constant, and must never contain any variable data from any origin. Do not be tempted to decide case-by-case whether an item of data is trusted, and continue using string concatenation within the query for cases that are considered safe. It is all too easy to make mistakes about the possible origin of data, or for changes in other code to violate assumptions about what data is tainted.
+* \[1] https://portswigger.net/burp
+* [2] https://portswigger.net/support/configuring-your-browser-to-work-with-burp
+* [3] https://portswigger.net/support/checking-your-browser-proxy-configuration
+* [4] https://portswigger.net/support/installing-burp-suites-ca-certificate-in-your-browser
+* [5] https://portswigger.net/web-security/sql-injection/cheat-sheet
+* [6] https://github.com/swisskyrepo/PayloadsAllTheThings/tree/master/SQL%20Injection
+* \[7] https://sqlmap.org/
```

Comment:
Why is there a backslash?

Reply:
It was something to use hyperlink but now I delete them and use simple hyperlinks for Markdown.